lib/galaxy/model/migrate/versions/0155_add_content_update_time.py

```diff
@@ -0,0 +1,102 @@
+"""
+Adds trigger to dataset table to update history.update_time when
+contents are changed.
+"""
+from __future__ import print_function
+
+import logging
+
+from sqlalchemy import DDL, MetaData
+
+log = logging.getLogger(__name__)
+metadata = MetaData()
+
+
+def upgrade(migrate_engine):
+    print(__doc__)
+    metadata.bind = migrate_engine
+    metadata.reflect()
+    drop_update_trigger(migrate_engine)
+    install_update_trigger(migrate_engine)
+
+
+def downgrade(migrate_engine):
+    metadata.bind = migrate_engine
+    metadata.reflect()
+    drop_update_trigger(migrate_engine)
+
+
+def install_update_trigger(migrate_engine):
+    """Installs trigger on database table to update history table
+    when contents have changed. Installs a function and a trigger
+    for postgres, other sql variants only require the trigger def
+    """
+
+    if migrate_engine.name in ['postgres', 'postgresql']:
+        pg_create_trigger = DDL("""
+            CREATE FUNCTION update_history_content_update_time()
+                RETURNS trigger
+                LANGUAGE 'plpgsql'
+            AS $BODY$
+            BEGIN
+                UPDATE history h
+                SET update_time = current_timestamp
+                FROM history_dataset_association hda
+                WHERE h.id = hda.history_id
+                AND hda.dataset_id = NEW.id;
+                RETURN NEW;
+            END;
+            $BODY$;
+            CREATE TRIGGER update_history_update_time
```

Comment:
I was thinking we may want to establish a trigger naming convention, what about `trigger_TABLE_OPERATION`? For this trigger that would be `trigger_dataset_aiudr`, where `aiudr` stands for "after insert, delete of update for each row".

Reply:
Okie dokie.